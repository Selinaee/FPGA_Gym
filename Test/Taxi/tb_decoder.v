`timescale 1ns / 1ps
`include "decoder.v"
module decode_tb;

  // Parameters

  //Ports
  reg clk;
  reg [8:0] encoded_state;
  wire [1:0] dest_idx;
  wire [2:0] pass_idx;
  wire [2:0] taxi_col;
  wire [2:0] taxi_row;

  decode  decode_inst (
    .encoded_state(encoded_state),
    .dest_idx(dest_idx),
    .pass_idx(pass_idx),
    .taxi_col(taxi_col),
    .taxi_row(taxi_row)
  );

  initial begin
    $dumpfile("tb_decoder.vcd");
    $dumpvars(0, decode_tb); // tb的模块名
  end

always #5  clk = ! clk ;
initial begin
  encoded_state = 9'b000000000;
  clk = 0;
  #10 encoded_state = 9'd211;
  #10
  $display("dest_idx = %d, pass_idx = %d, taxi_col = %d, taxi_row = %d", dest_idx, pass_idx, taxi_col, taxi_row);
  #1000 $finish;
end

endmodule