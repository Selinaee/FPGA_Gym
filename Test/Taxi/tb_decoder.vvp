#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\system.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\va_math.vpi";
S_000001e5cc46bb70 .scope module, "decode_tb" "decode_tb" 2 3;
 .timescale -9 -12;
v000001e5cc482d70_0 .var "clk", 0 0;
v000001e5cc482e10_0 .net "dest_idx", 1 0, v000001e5cc46b8e0_0;  1 drivers
v000001e5cc482eb0_0 .var "encoded_state", 8 0;
v000001e5cc482f50_0 .net "pass_idx", 2 0, v000001e5cc482b90_0;  1 drivers
v000001e5cc4a5560_0 .net "taxi_col", 2 0, v000001e5cc482c30_0;  1 drivers
v000001e5cc4a5600_0 .net "taxi_row", 2 0, v000001e5cc482cd0_0;  1 drivers
S_000001e5cc46bd00 .scope module, "decode_inst" "decode" 2 15, 3 26 0, S_000001e5cc46bb70;
 .timescale -9 -12;
    .port_info 0 /INPUT 9 "encoded_state";
    .port_info 1 /OUTPUT 2 "dest_idx";
    .port_info 2 /OUTPUT 3 "pass_idx";
    .port_info 3 /OUTPUT 3 "taxi_col";
    .port_info 4 /OUTPUT 3 "taxi_row";
v000001e5cc46b8e0_0 .var "dest_idx", 1 0;
v000001e5cc46be90_0 .net "encoded_state", 8 0, v000001e5cc482eb0_0;  1 drivers
v000001e5cc46bf30_0 .var/i "intermediate", 31 0;
v000001e5cc482b90_0 .var "pass_idx", 2 0;
v000001e5cc482c30_0 .var "taxi_col", 2 0;
v000001e5cc482cd0_0 .var "taxi_row", 2 0;
E_000001e5cc4b7d20 .event anyedge, v000001e5cc46be90_0, v000001e5cc46bf30_0;
    .scope S_000001e5cc46bd00;
T_0 ;
    %wait E_000001e5cc4b7d20;
    %load/vec4 v000001e5cc46be90_0;
    %pad/u 32;
    %pushi/vec4 4, 0, 32;
    %mod;
    %pad/u 2;
    %store/vec4 v000001e5cc46b8e0_0, 0, 2;
    %load/vec4 v000001e5cc46be90_0;
    %pad/u 32;
    %pushi/vec4 4, 0, 32;
    %div;
    %store/vec4 v000001e5cc46bf30_0, 0, 32;
    %load/vec4 v000001e5cc46bf30_0;
    %pushi/vec4 5, 0, 32;
    %mod/s;
    %pad/s 3;
    %store/vec4 v000001e5cc482b90_0, 0, 3;
    %load/vec4 v000001e5cc46bf30_0;
    %pushi/vec4 5, 0, 32;
    %div/s;
    %store/vec4 v000001e5cc46bf30_0, 0, 32;
    %load/vec4 v000001e5cc46bf30_0;
    %pushi/vec4 5, 0, 32;
    %mod/s;
    %pad/s 3;
    %store/vec4 v000001e5cc482c30_0, 0, 3;
    %load/vec4 v000001e5cc46bf30_0;
    %pushi/vec4 5, 0, 32;
    %div/s;
    %store/vec4 v000001e5cc46bf30_0, 0, 32;
    %load/vec4 v000001e5cc46bf30_0;
    %pad/s 3;
    %store/vec4 v000001e5cc482cd0_0, 0, 3;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_000001e5cc46bb70;
T_1 ;
    %vpi_call 2 24 "$dumpfile", "tb_decoder.vcd" {0 0 0};
    %vpi_call 2 25 "$dumpvars", 32'sb00000000000000000000000000000000, S_000001e5cc46bb70 {0 0 0};
    %end;
    .thread T_1;
    .scope S_000001e5cc46bb70;
T_2 ;
    %delay 5000, 0;
    %load/vec4 v000001e5cc482d70_0;
    %nor/r;
    %store/vec4 v000001e5cc482d70_0, 0, 1;
    %jmp T_2;
    .thread T_2;
    .scope S_000001e5cc46bb70;
T_3 ;
    %pushi/vec4 0, 0, 9;
    %store/vec4 v000001e5cc482eb0_0, 0, 9;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001e5cc482d70_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 211, 0, 9;
    %store/vec4 v000001e5cc482eb0_0, 0, 9;
    %delay 10000, 0;
    %vpi_call 2 34 "$display", "dest_idx = %d, pass_idx = %d, taxi_col = %d, taxi_row = %d", v000001e5cc482e10_0, v000001e5cc482f50_0, v000001e5cc4a5560_0, v000001e5cc4a5600_0 {0 0 0};
    %delay 1000000, 0;
    %vpi_call 2 35 "$finish" {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "tb_decoder.v";
    "./decoder.v";
