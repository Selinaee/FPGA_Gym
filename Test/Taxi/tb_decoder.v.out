#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\system.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "E:\Software2\iverilog\iverilog\lib\ivl\va_math.vpi";
S_000001d7e954ddc0 .scope module, "decode_tb" "decode_tb" 2 3;
 .timescale -9 -12;
v000001d7e9604990_0 .var "clk", 0 0;
v000001d7e9604a30_0 .net "dest_idx", 1 0, v000001d7e9606d80_0;  1 drivers
v000001d7e9604ad0_0 .var "encoded_state", 8 0;
v000001d7e9604b70_0 .net "pass_idx", 2 0, v000001d7e9606ec0_0;  1 drivers
v000001d7e9604c10_0 .net "taxi_col", 2 0, v000001d7e9606f60_0;  1 drivers
v000001d7e9604cb0_0 .net "taxi_row", 2 0, v000001d7e96048f0_0;  1 drivers
S_000001d7e9606bf0 .scope module, "decode_inst" "decode" 2 15, 3 3 0, S_000001d7e954ddc0;
 .timescale -9 -12;
    .port_info 0 /INPUT 9 "encoded_state";
    .port_info 1 /OUTPUT 2 "dest_idx";
    .port_info 2 /OUTPUT 3 "pass_idx";
    .port_info 3 /OUTPUT 3 "taxi_col";
    .port_info 4 /OUTPUT 3 "taxi_row";
v000001d7e9606d80_0 .var "dest_idx", 1 0;
v000001d7e9543980_0 .net "encoded_state", 8 0, v000001d7e9604ad0_0;  1 drivers
v000001d7e9606e20_0 .var/i "intermediate", 31 0;
v000001d7e9606ec0_0 .var "pass_idx", 2 0;
v000001d7e9606f60_0 .var "taxi_col", 2 0;
v000001d7e96048f0_0 .var "taxi_row", 2 0;
E_000001d7e9609be0 .event anyedge, v000001d7e9543980_0, v000001d7e9606e20_0;
    .scope S_000001d7e9606bf0;
T_0 ;
    %wait E_000001d7e9609be0;
    %load/vec4 v000001d7e9543980_0;
    %pad/u 32;
    %pushi/vec4 4, 0, 32;
    %mod;
    %pad/u 2;
    %store/vec4 v000001d7e9606d80_0, 0, 2;
    %load/vec4 v000001d7e9543980_0;
    %pad/u 32;
    %pushi/vec4 4, 0, 32;
    %div;
    %store/vec4 v000001d7e9606e20_0, 0, 32;
    %load/vec4 v000001d7e9606e20_0;
    %pushi/vec4 5, 0, 32;
    %mod/s;
    %pad/s 3;
    %store/vec4 v000001d7e9606ec0_0, 0, 3;
    %load/vec4 v000001d7e9606e20_0;
    %pushi/vec4 5, 0, 32;
    %div/s;
    %store/vec4 v000001d7e9606e20_0, 0, 32;
    %load/vec4 v000001d7e9606e20_0;
    %pushi/vec4 5, 0, 32;
    %mod/s;
    %pad/s 3;
    %store/vec4 v000001d7e9606f60_0, 0, 3;
    %load/vec4 v000001d7e9606e20_0;
    %pushi/vec4 5, 0, 32;
    %div/s;
    %store/vec4 v000001d7e9606e20_0, 0, 32;
    %load/vec4 v000001d7e9606e20_0;
    %pad/s 3;
    %store/vec4 v000001d7e96048f0_0, 0, 3;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_000001d7e954ddc0;
T_1 ;
    %vpi_call 2 24 "$dumpfile", "tb_decoder.vcd" {0 0 0};
    %vpi_call 2 25 "$dumpvars", 32'sb00000000000000000000000000000000, S_000001d7e954ddc0 {0 0 0};
    %end;
    .thread T_1;
    .scope S_000001d7e954ddc0;
T_2 ;
    %delay 5000, 0;
    %load/vec4 v000001d7e9604990_0;
    %nor/r;
    %store/vec4 v000001d7e9604990_0, 0, 1;
    %jmp T_2;
    .thread T_2;
    .scope S_000001d7e954ddc0;
T_3 ;
    %pushi/vec4 0, 0, 9;
    %store/vec4 v000001d7e9604ad0_0, 0, 9;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001d7e9604990_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 211, 0, 9;
    %store/vec4 v000001d7e9604ad0_0, 0, 9;
    %delay 10000, 0;
    %vpi_call 2 34 "$display", "dest_idx = %d, pass_idx = %d, taxi_col = %d, taxi_row = %d", v000001d7e9604a30_0, v000001d7e9604b70_0, v000001d7e9604c10_0, v000001d7e9604cb0_0 {0 0 0};
    %delay 1000000, 0;
    %vpi_call 2 35 "$finish" {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "tb_decoder.v";
    "./decoder.v";
